composer/callbacks/global_straggler_detector.py

```diff
@@ -0,0 +1,873 @@
+            self.logger.warning(f"{str(ex_val)}\n{err}")
+            ret = True
+        self.stop()
+        return ret
+
+
+
+class GlobalStragglerDetector(Callback):
+    """Logs the minimum and maximum training values across all ranks for the following metrics:
+
+        Rtt : RoundTrip Time (time spent in all the traced ops in the current batch)
+        Pwr : GPU Power
+        Tmp : GPU Temperature
+        Utl : GPU Utilization
+        Clk : GPU Clock
+        DRtt: Batch loading latency (time spent loading the current batch from the dataset)
+        Etpt: Estimated throughput for the current batch
+
+    The maximum and minimum values for these metrics, alongside their respective ranks, are logged 
+    on the :attr:`.Event.BATCH_END` event for every batch. 
+
+    To compute `flops_per_sec`, the model attribute `flops_per_batch` should be set to a callable
+    which accepts a batch and returns the number of flops for that batch. Typically, this should
+    be flops per sample times the batch size unless pad tokens are used.
+
+    The wall clock time is logged on every :attr:`.Event.BATCH_END` event.
+
+    Example:
+        .. doctest::
+
+            >>> from composer import Trainer
+            >>> from composer.callbacks import GlobalStragglerDetector
+            >>> # constructing trainer object with this callback
+            >>> trainer = Trainer(
+            ...     model=model,
+            ...     train_dataloader=train_dataloader,
+            ...     eval_dataloader=eval_dataloader,
+            ...     optimizers=optimizer,
+            ...     max_duration='1ep',
+            ...     callbacks=[GlobalStragglerDetector()],
+            ... )
+
+    The metrics are logged by the :class:`.Logger` to the following keys as
+    described below.
+
+    +-------------------------------------+-----------------------------------------------------------+
+    | Key                                 | Logged data                                               |
+    +=====================================+===========================================================+
+    |                                     | Minimum time spent in all the traced ops in the           |
+    | `MnRtt/Rnk`                         | current batch across all ranks/corresponding rank         |
+    |                                     |                                                           |
+    +-------------------------------------+-----------------------------------------------------------+
+    |                                     | Maximum time spent in all the traced ops in the           |
+    | `MxRtt/Rnk`                         | current batch across all ranks/corresponding rank         |
+    |                                     |                                                           |
+    +-------------------------------------+-----------------------------------------------------------+
```

Comment:
May consider fully writing out the acronym's and specifying the rank in the key itself?